lib/src/database/matrix_sdk_database.dart

```diff
@@ -1618,4 +1627,31 @@ class MatrixSdkDatabase extends DatabaseApi with DatabaseFileStorage {
         name,
         sqfliteFactory ?? idbFactory,
       );
+
+  @override
+  Future<void> markUserProfileAsOutdated(userId) async {
+    final profile = await getUserProfile(userId);
+    if (profile == null) return;
+    await _userProfilesBox.put(
+      userId,
+      CachedProfileInformation.fromProfile(
+        profile as ProfileInformation,
+        outdated: true,
+        updated: profile.updated,
+      ).toJson(),
+    );
+  }
+
+  @override
+  Future<CachedProfileInformation?> getUserProfile(String userId) =>
+      _userProfilesBox.get(userId).then((json) => json == null
+          ? null
+          : CachedProfileInformation.fromJson(copyMap(json)));
+
+  @override
+  Future<void> storeUserProfile(String userId, ProfileInformation profile) =>
```

Comment:
shouldn't the `profile` here be `CachedProfileInformation` too?

Reply:
Ah yes you are right. Thanks. Have changed it